icu4c/source/common/locid.cpp

```diff
@@ -904,6 +920,9 @@ AliasDataBuilder::build(UErrorCode &status) {
     LocalUResourceBundlePointer variantAlias(
         ures_getByKey(metadataAlias.getAlias(), "variant", nullptr, &status));
 
+    if (U_FAILURE(status)) {
+        return nullptr;
+    }
     int32_t languagesLength = 0, scriptLength = 0, territoryLength = 0,
             variantLength = 0;
 
```

Comment:
(Note: GitHub won't let me comment directly on a line far from where you have made changes already.)

Please do change the end of this build() function from
```
    // copy hashtables
    return new AliasData(
        std::move(languageMap),
        std::move(scriptMap),
        std::move(territoryMap),
        std::move(variantMap),
        strings.orphanCharStrings());
```
to
```
    if (U_FAILURE(status)) {
        return nullptr;
    }

    // copy hashtables
    auto *data = new AliasData(
        std::move(languageMap),
        std::move(scriptMap),
        std::move(territoryMap),
        std::move(variantMap),
        strings.orphanCharStrings());
    if (data == nullptr) {
        status = U_MEMORY_ALLOCATION_ERROR;
    }
    return data;
```